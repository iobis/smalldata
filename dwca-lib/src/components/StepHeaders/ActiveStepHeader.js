import PropTypes from 'prop-types'
import React from 'react'
import StepFooter from './StepFooter'
import StepHeader from './StepHeader'

export default function ActiveStepHeader({
  activeStepIndex,
  children,
  nextStep,
  onContinueButtonClick,
  onCopyPreviousDataClick,
  onStepTitleClick,
  stepDescription,
  stepTitle,
  totalSteps
}) {
  const footerVisible = activeStepIndex <= totalSteps
  return (
    <>
      <StepHeader
        className="active has-background-grey has-text-white"
        dataDescription=""
        iconVisible={false}
        onStepTitleClick={onStepTitleClick}
        scrollOnRender={true}
        selectedData=""
        stepDescription={stepDescription}
        stepTitle={stepTitle}>
        {children}
      </StepHeader>
      {footerVisible && (
        <StepFooter
          activeStepIndex={activeStepIndex}
          nextStep={nextStep}
          onContinueButtonClick={onContinueButtonClick}
          onCopyPreviousDataClick={onCopyPreviousDataClick}/>
      )}
    </>
  )
}

ActiveStepHeader.propTypes = {
  activeStepIndex:         PropTypes.number.isRequired,
  children:                PropTypes.node,
  nextStep:                PropTypes.string,
  onContinueButtonClick:   PropTypes.func.isRequired,
  onCopyPreviousDataClick: PropTypes.func,
  onStepTitleClick:        PropTypes.func.isRequired,
  stepDescription:         PropTypes.string.isRequired,
  stepTitle:               PropTypes.string.isRequired,
  totalSteps:              PropTypes.number.isRequired
}
