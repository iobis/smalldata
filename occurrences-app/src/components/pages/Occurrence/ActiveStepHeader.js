import PropTypes from 'prop-types'
import React from 'react'
import StepHeader from './StepHeader'

export default function ActiveStepHeader({ stepDescription, stepTitle }) {
  return (
    <StepHeader
      className="active has-background-grey has-text-white"
      dataDescription=""
      iconVisible={false}
      selectedData=""
      stepDescription={stepDescription}
      stepTitle={stepTitle}/>
  )
}

ActiveStepHeader.propTypes = {
  stepDescription: PropTypes.string.isRequired,
  stepTitle:       PropTypes.string.isRequired
}
