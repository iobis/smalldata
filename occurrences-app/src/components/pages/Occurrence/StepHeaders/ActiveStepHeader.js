import PropTypes from 'prop-types'
import React from 'react'
import StepHeader from './StepHeader'
import CopyPreviousData from '../CopyPreviousData'
import ContinueButton from '../ContinueButton'

export default function ActiveStepHeader({ children, onStepTitleClick, stepDescription, stepTitle, onContinueButtonClick, totalSteps, activeStepIndex }) {
  return (
    <>
      <StepHeader
        className="active has-background-grey has-text-white"
        dataDescription=""
        iconVisible={false}
        onStepTitleClick={onStepTitleClick}
        selectedData=""
        stepDescription={stepDescription}
        stepTitle={stepTitle}>
        {children}
      </StepHeader>
      {activeStepIndex < totalSteps &&
      <div className="columns obis-spaced">
        {activeStepIndex > 0 &&
        <CopyPreviousData/>
        }
        <ContinueButton
          name="locationContinue"
          onClick={onContinueButtonClick}
          value="occurrenceForm.locationData.step.nextStep"
          wrapperClassName=""/>
      </div>
      }
    </>
  )
}

ActiveStepHeader.propTypes = {
  activeStepIndex:       PropTypes.number,
  children:              PropTypes.node,
  onContinueButtonClick: PropTypes.func.isRequired,
  onStepTitleClick:      PropTypes.func.isRequired,
  stepDescription:       PropTypes.string.isRequired,
  stepTitle:             PropTypes.string.isRequired,
  totalSteps:            PropTypes.number
}
