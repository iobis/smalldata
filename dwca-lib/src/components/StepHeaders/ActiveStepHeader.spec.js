import ActiveStepHeader from './ActiveStepHeader'
import React from 'react'
import { mount } from 'enzyme'

describe('ActiveStepHeader', () => {
  it('renders StepFooter with all data if next step is provided', () => {
    const wrapper = mount(
      <ActiveStepHeader
        activeStepIndex={1}
        nextStep="next step"
        onContinueButtonClick={() => {}}
        onCopyPreviousDataClick={() => {}}
        onStepTitleClick={() => {}}
        stepDescription="step description"
        stepTitle="step title"
        totalSteps={6}/>
    )

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('.step-footer').exists()).toBe(true)
    expect(wrapper.find('.step-footer .copy-previous-data').exists()).toBe(true)
    expect(wrapper.find('.step-footer .copy-previous-data').text()).toBe('occurrenceForm.copyPreviousStep')
    expect(wrapper.find('.step-footer .continue-button').exists()).toBe(true)
  })

  it('does not render StepFooter if next step is not provided', () => {
    const wrapper = mount(
      <ActiveStepHeader
        activeStepIndex={1}
        onContinueButtonClick={() => {}}
        onCopyPreviousDataClick={() => {}}
        onStepTitleClick={() => {}}
        stepDescription="step description"
        stepTitle="step title"
        totalSteps={6}/>
    )

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('.step-footer').exists()).toBe(true)
    expect(wrapper.find('.step-footer .copy-previous-data').exists()).toBe(true)
    expect(wrapper.find('.step-footer .copy-previous-data').text()).toBe('occurrenceForm.copyPreviousStep')
    expect(wrapper.find('.step-footer .continue-button').exists()).toBe(false)
  })

  it('does not render StepFooter if last step', () => {
    const wrapper = mount(
      <ActiveStepHeader
        activeStepIndex={6}
        onContinueButtonClick={() => {}}
        onCopyPreviousDataClick={() => {}}
        onStepTitleClick={() => {}}
        stepDescription="step description"
        stepTitle="step title"
        totalSteps={6}/>
    )

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('.step-footer').exists()).toBe(true)
    expect(wrapper.find('.step-footer .copy-previous-data').exists()).toBe(true)
    expect(wrapper.find('.step-footer .copy-previous-data').text()).toBe('occurrenceForm.copyPreviousStep')
    expect(wrapper.find('.step-footer .continue-button').exists()).toBe(false)
  })

  it('does not render copy previous step if onCopyPreviousDataClick not provided', () => {
    const wrapper = mount(
      <ActiveStepHeader
        activeStepIndex={1}
        onContinueButtonClick={() => {}}
        onStepTitleClick={() => {}}
        stepDescription="step description"
        stepTitle="step title"
        totalSteps={6}/>
    )

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('.step-footer').exists()).toBe(true)
    expect(wrapper.find('.step-footer .copy-previous-data').exists()).toBe(true)
    expect(wrapper.find('.step-footer .copy-previous-data').text()).toBe('')
    expect(wrapper.find('.step-footer .continue-button').exists()).toBe(false)
  })
})
