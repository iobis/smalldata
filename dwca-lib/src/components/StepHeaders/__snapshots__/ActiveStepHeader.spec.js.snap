// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActiveStepHeader does not render StepFooter if last step 1`] = `
<ActiveStepHeader
  activeStepIndex={6}
  onContinueButtonClick={[Function]}
  onCopyPreviousDataClick={[Function]}
  onStepTitleClick={[Function]}
  stepDescription="step description"
  stepTitle="step title"
  totalSteps={6}
>
  <StepHeader
    className="active has-background-grey has-text-white"
    dataDescription=""
    iconVisible={false}
    onStepTitleClick={[Function]}
    selectedData=""
    stepDescription="step description"
    stepTitle="step title"
  >
    <header
      className="step-header columns is-vcentered active has-background-grey has-text-white"
      onClick={[Function]}
    >
      <div
        className="column is-1 is-hidden-mobile"
      >
        <p
          className="datadescription"
        />
      </div>
      <p
        className="selected-data column is-hidden-mobile"
      />
      <div
        className="column details"
      >
        <h3
          className="is-size-6"
        >
          step description
        </h3>
        <h2
          className="step-title"
        >
          step title
        </h2>
      </div>
    </header>
  </StepHeader>
  <StepFooter
    activeStepIndex={6}
    onContinueButtonClick={[Function]}
    onCopyPreviousDataClick={[Function]}
  >
    <div
      className="step-footer columns section"
    >
      <CopyPreviousData
        onClick={[Function]}
        visible={true}
      >
        <div
          className="columns"
        >
          <div
            className="copy-previous-data column is-narrow"
            onClick={[Function]}
          >
            occurrenceForm.copyPreviousStep
          </div>
        </div>
      </CopyPreviousData>
    </div>
  </StepFooter>
</ActiveStepHeader>
`;

exports[`ActiveStepHeader does not render StepFooter if next step is not provided 1`] = `
<ActiveStepHeader
  activeStepIndex={1}
  onContinueButtonClick={[Function]}
  onCopyPreviousDataClick={[Function]}
  onStepTitleClick={[Function]}
  stepDescription="step description"
  stepTitle="step title"
  totalSteps={6}
>
  <StepHeader
    className="active has-background-grey has-text-white"
    dataDescription=""
    iconVisible={false}
    onStepTitleClick={[Function]}
    selectedData=""
    stepDescription="step description"
    stepTitle="step title"
  >
    <header
      className="step-header columns is-vcentered active has-background-grey has-text-white"
      onClick={[Function]}
    >
      <div
        className="column is-1 is-hidden-mobile"
      >
        <p
          className="datadescription"
        />
      </div>
      <p
        className="selected-data column is-hidden-mobile"
      />
      <div
        className="column details"
      >
        <h3
          className="is-size-6"
        >
          step description
        </h3>
        <h2
          className="step-title"
        >
          step title
        </h2>
      </div>
    </header>
  </StepHeader>
  <StepFooter
    activeStepIndex={1}
    onContinueButtonClick={[Function]}
    onCopyPreviousDataClick={[Function]}
  >
    <div
      className="step-footer columns section"
    >
      <CopyPreviousData
        onClick={[Function]}
        visible={true}
      >
        <div
          className="columns"
        >
          <div
            className="copy-previous-data column is-narrow"
            onClick={[Function]}
          >
            occurrenceForm.copyPreviousStep
          </div>
        </div>
      </CopyPreviousData>
    </div>
  </StepFooter>
</ActiveStepHeader>
`;

exports[`ActiveStepHeader does not render copy previous step if onCopyPreviousDataClick not provided 1`] = `
<ActiveStepHeader
  activeStepIndex={1}
  onContinueButtonClick={[Function]}
  onStepTitleClick={[Function]}
  stepDescription="step description"
  stepTitle="step title"
  totalSteps={6}
>
  <StepHeader
    className="active has-background-grey has-text-white"
    dataDescription=""
    iconVisible={false}
    onStepTitleClick={[Function]}
    selectedData=""
    stepDescription="step description"
    stepTitle="step title"
  >
    <header
      className="step-header columns is-vcentered active has-background-grey has-text-white"
      onClick={[Function]}
    >
      <div
        className="column is-1 is-hidden-mobile"
      >
        <p
          className="datadescription"
        />
      </div>
      <p
        className="selected-data column is-hidden-mobile"
      />
      <div
        className="column details"
      >
        <h3
          className="is-size-6"
        >
          step description
        </h3>
        <h2
          className="step-title"
        >
          step title
        </h2>
      </div>
    </header>
  </StepHeader>
  <StepFooter
    activeStepIndex={1}
    onContinueButtonClick={[Function]}
  >
    <div
      className="step-footer columns section"
    >
      <CopyPreviousData
        visible={false}
      >
        <div
          className="columns"
        >
          <div
            className="copy-previous-data column is-narrow"
          />
        </div>
      </CopyPreviousData>
    </div>
  </StepFooter>
</ActiveStepHeader>
`;

exports[`ActiveStepHeader renders StepFooter with all data if next step is provided 1`] = `
<ActiveStepHeader
  activeStepIndex={1}
  nextStep="next step"
  onContinueButtonClick={[Function]}
  onCopyPreviousDataClick={[Function]}
  onStepTitleClick={[Function]}
  stepDescription="step description"
  stepTitle="step title"
  totalSteps={6}
>
  <StepHeader
    className="active has-background-grey has-text-white"
    dataDescription=""
    iconVisible={false}
    onStepTitleClick={[Function]}
    selectedData=""
    stepDescription="step description"
    stepTitle="step title"
  >
    <header
      className="step-header columns is-vcentered active has-background-grey has-text-white"
      onClick={[Function]}
    >
      <div
        className="column is-1 is-hidden-mobile"
      >
        <p
          className="datadescription"
        />
      </div>
      <p
        className="selected-data column is-hidden-mobile"
      />
      <div
        className="column details"
      >
        <h3
          className="is-size-6"
        >
          step description
        </h3>
        <h2
          className="step-title"
        >
          step title
        </h2>
      </div>
    </header>
  </StepHeader>
  <StepFooter
    activeStepIndex={1}
    nextStep="next step"
    onContinueButtonClick={[Function]}
    onCopyPreviousDataClick={[Function]}
  >
    <div
      className="step-footer columns section"
    >
      <CopyPreviousData
        onClick={[Function]}
        visible={true}
      >
        <div
          className="columns"
        >
          <div
            className="copy-previous-data column is-narrow"
            onClick={[Function]}
          >
            occurrenceForm.copyPreviousStep
          </div>
        </div>
      </CopyPreviousData>
      <ContinueButton
        name="locationContinue"
        onClick={[Function]}
        value="next step"
      >
        <button
          className="continue-button button is-info"
          name="locationContinue"
          onClick={[Function]}
        >
          common.continueTo next step
        </button>
      </ContinueButton>
    </div>
  </StepFooter>
</ActiveStepHeader>
`;
